lib/css_parser/parser.rb

```diff
@@ -277,7 +277,7 @@ def parse_block_into_rule_sets!(block, options = {}) # :nodoc:
             next
           end
 
-          if token =~ /\{/
+          if token =~ /\{/ and not in_string
```

Comment:
The solution, then, is to treat the open bracket the same as the closed bracket.